opm/output/eclipse/EclipseWriter.cpp

```diff
@@ -104,171 +77,98 @@ void restrictAndReorderToActiveCells(std::vector<double> &data,
-        }
+        for( size_t i = 0; i < data.size(); ++i )
+            target[ i ] = T( data[ i ] );
     }
 
-    void set(const std::string name, const std::vector<const char *>& data)
+    Keyword( const std::string& name, const std::vector< const char* >& data ) :
+        handle( ecl_kw_alloc( name.c_str(), data.size(), ECL_CHAR_TYPE ) )
     {
-      if(ertHandle_) {
-          ecl_kw_free(ertHandle_);
-      }
-
-
-      ertHandle_ = ecl_kw_alloc(name.c_str(),
-                                data.size(),
-                                ertType_());
-
-      // number of elements to take
-      const int numEntries = data.size();
-      for (int i = 0; i < numEntries; ++i) {
-          ecl_kw_iset_char_ptr( ertHandle_, i, data[i]);
-     }
-
+        for( size_t i = 0; i < data.size(); ++i )
+          ecl_kw_iset_char_ptr( this->handle.get(), i, data[ i ] );
     }
 
-    ecl_kw_type *ertHandle() const
-    { return ertHandle_; }
+    ecl_kw_type *ertHandle() { return this->handle.get(); }
+    const ecl_kw_type *ertHandle() const { return this->handle.get(); }
 
 private:
-    static ecl_type_enum ertType_()
-    {
-        if (std::is_same<T, float>::value)
-        { return ECL_FLOAT_TYPE; }
-        if (std::is_same<T, double>::value)
-        { return ECL_DOUBLE_TYPE; }
-        if (std::is_same<T, int>::value)
-        { return ECL_INT_TYPE; }
-        if (std::is_same<T, const char *>::value)
-        { return ECL_CHAR_TYPE; }
-
-
-        OPM_THROW(std::logic_error,
-                  "Unhandled type for data elements in EclipseWriterDetails::Keyword");
-    }
-
-    ecl_kw_type *ertHandle_;
+    ERT::ert_unique_ptr< ecl_kw_type, ecl_kw_free > handle;
 };
 
 /**
  * Pointer to memory that holds the name to an Eclipse output file.
  */
-class FileName : private boost::noncopyable
-{
+class FileName {
 public:
     FileName(const std::string& outputDir,
              const std::string& baseName,
              ecl_file_enum type,
              int writeStepIdx,
-             bool formatted)
-    {
-        ertHandle_ = ecl_util_alloc_filename(outputDir.c_str(),
-                                             baseName.c_str(),
-                                             type,
-                                             formatted,
-                                             writeStepIdx);
-    }
-
-    ~FileName()
-    { std::free(ertHandle_); }
+             bool formatted ) :
+        handle( ecl_util_alloc_filename(
+                                outputDir.c_str(),
+                                baseName.c_str(),
+                                type,
+                                formatted,
+                                writeStepIdx ),
+                std::free )
+    {}
 
-    const char *ertHandle() const
-    { return ertHandle_; }
+    const char* ertHandle() const { return this->handle.get(); }
 
 private:
-    char *ertHandle_;
+    using fd = std::unique_ptr< char, decltype( std::free )* >;
```

Comment:
`file descriptor` as is common in linux programs. Anywho, it was mostly to make the variable declaration neater, and no biggie.